components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java

```diff
@@ -0,0 +1,179 @@
+            LOG.error("Unable to create " + TEMP_DIR, e);
+        }
+
+    }
+
+    MiniSolrCloudCluster miniCluster;
+    File testDir;
+    SolrZkClient zkClient;
+
+    CloudSolrClient solrClient;
+
+    public SolrCloudFixture(String solrHome) throws Exception {
+        String xml = IOHelper.loadText(new FileInputStream(new File(solrHome, "solr-no-core.xml")));
+        miniCluster = new MiniSolrCloudCluster(1, "/solr", TEMP_DIR, xml, null, null);
+        String zkAddr = miniCluster.getZkServer().getZkAddress();
+        String zkHost = miniCluster.getZkServer().getZkHost();
+
+        buildZooKeeper(zkHost, zkAddr, new File(solrHome), "solrconfig.xml", "schema.xml");
+        List<JettySolrRunner> jettys = miniCluster.getJettySolrRunners();
+        for (JettySolrRunner jetty : jettys) {
+            if (!jetty.isRunning()) {
+                LOG.warn("JETTY NOT RUNNING!");
+            } else {
+                LOG.info("JETTY RUNNING AT " + jetty.getBaseUrl() + " PORT " + jetty.getLocalPort());
+            }
+        }
+
+        solrClient = new CloudHttp2SolrClient.Builder(Arrays.asList(zkAddr), Optional.empty())
+                .withDefaultCollection("collection1")
+                .build();
+        solrClient.connect();
+
+        createCollection(solrClient, "collection1", 1, 1, "conf1");
+        Thread.sleep(1000); // takes some time to setup the collection...
+                           // otherwise you'll get no live solr servers
+
+        SolrInputDocument doc = new SolrInputDocument();
+        doc.setField("id", "1");
+
+        solrClient.add(doc);
+        solrClient.commit();
+    }
+
+    public static void putConfig(String confName, SolrZkClient zkClient, File solrhome, final String name)
+            throws Exception {
+        putConfig(confName, zkClient, solrhome, name, name);
+    }
+
+    protected NamedList<Object> createCollection(
+            CloudSolrClient server, String name, int numShards,
+            int replicationFactor, String configName)
+            throws Exception {
+        ModifiableSolrParams modParams = new ModifiableSolrParams();
+        modParams.set(CoreAdminParams.ACTION, CollectionAction.CREATE.name());
+        modParams.set("name", name);
+        modParams.set("numShards", numShards);
+        modParams.set("replicationFactor", replicationFactor);
+        modParams.set("collection.configName", configName);
+        QueryRequest request = new QueryRequest(modParams);
+        request.setPath("/admin/collections");
+        return server.request(request);
+    }
+
+    public static void putConfig(
+            String confName, SolrZkClient zkClient, File solrhome, final String srcName,
+            String destName)
+            throws Exception {
+        File file = new File(solrhome, "collection1" + File.separator + "conf" + File.separator + srcName);
+        if (!file.exists()) {
+            LOG.info("zk skipping " + file.getAbsolutePath() + " because it doesn't exist");
```

Comment:
Log placeholders here and in all other places.

Reply:
fixed